### ISSUE TITLE
Update GHActions/JamesIves/github-pages-deploy-action to 4.0.0 for all repos

### ISSUE THREAD

**Issue by himorin** [immersive-web/administrivia#151]
[4.0.0 has released](https://github.com/JamesIves/github-pages-deploy-action/releases/tag/4.0.0) and release note notes that GITHUB_TOKEN is no longer required. So, let me propose to update all repositories with 4.0.0 and removing GITHUB_TOKEN.

Test worked well at my fork: https://github.com/himorin/webxr/runs/1852857068

/cc @Yonet @AdaRoseCannon @cwilso

**Comment by himorin** [comment]
Since there is no repository specific difference over bikeshed repositories, I will run [script](https://github.com/himorin/w3c-memo/blob/main/scripts/update-ghactions.sh) shortly (targeting Wed) - tested with my fork and working fine.

**Comment by Yonet** [comment]
@himorin thank you!

**Comment by himorin** [comment]
done.

(log)
```
% ./update-ghactions.sh
Directory . is not git repository
Working on ./WebXR-WebGPU-Binding, hit ENTER to go!:
./WebXR-WebGPU-Binding does not have GHActions directory
Working on ./administrivia, hit ENTER to go!:
./administrivia does not have GHActions directory
Working on ./anchors, hit ENTER to go!:
[main b959847] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 224.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/anchors
   364164f..b959847  main -> main
Finished for ./anchors
Working on ./ar-common, hit ENTER to go!:
./ar-common does not have GHActions directory
Working on ./cardboard-vr-display, hit ENTER to go!:
./cardboard-vr-display does not have GHActions directory
Working on ./community-resources, hit ENTER to go!:
./community-resources does not have GHActions directory
Working on ./computer-vision, hit ENTER to go!:
./computer-vision does not have GHActions directory
Working on ./depth-sensing, hit ENTER to go!:
[main e03cfc5] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 4 insertions(+), 3 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.19 KiB | 243.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/depth-sensing
   c67b280..e03cfc5  main -> main
Finished for ./depth-sensing
Working on ./detached-elements, hit ENTER to go!:
./detached-elements does not have GHActions directory
Working on ./dom-overlays, hit ENTER to go!:
[main f2d26f0] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/dom-overlays
   9138f66..f2d26f0  main -> main
Finished for ./dom-overlays
Working on ./geo-alignment, hit ENTER to go!: 
./geo-alignment does not have GHActions directory
Working on ./hit-test, hit ENTER to go!: 
[main 4ba0083] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/hit-test
   0cec518..4ba0083  main -> main
Finished for ./hit-test
Working on ./homepage, hit ENTER to go!: 
./homepage does not have GHActions directory
Working on ./immersive-web, hit ENTER to go!: 
./immersive-web does not have GHActions directory
Working on ./immersive-web-weekly, hit ENTER to go!: 
./immersive-web-weekly does not have GHActions directory
Working on ./immersive-web.github.io, hit ENTER to go!: 
./immersive-web.github.io does not have GHActions directory
Working on ./immersiveweb.dev, hit ENTER to go!: 
./immersiveweb.dev does not have GHActions directory
Working on ./layers, hit ENTER to go!: 
[main dd23af1] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 160.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/layers
   9c5652a..dd23af1  main -> main
Finished for ./layers
Working on ./lighting-estimation, hit ENTER to go!: 
[main 22ea002] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 160.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/lighting-estimation
   6810173..22ea002  main -> main
Finished for ./lighting-estimation
Working on ./marker-tracking, hit ENTER to go!: 
./marker-tracking does not have GHActions directory
Working on ./navigation, hit ENTER to go!: 
./navigation does not have GHActions directory
Working on ./occlusion, hit ENTER to go!: 
./occlusion does not have GHActions directory
Working on ./performance-improvements, hit ENTER to go!: 
./performance-improvements does not have GHActions directory
Working on ./privacy-and-security, hit ENTER to go!: 
./privacy-and-security does not have GHActions directory
Working on ./proposals, hit ENTER to go!: 
./proposals does not have GHActions directory
Working on ./real-world-geometry, hit ENTER to go!: 
./real-world-geometry does not have GHActions directory
Working on ./spatial-favicons, hit ENTER to go!: 
./spatial-favicons does not have GHActions directory
Working on ./webvr, hit ENTER to go!: 
./webvr does not have GHActions directory
Working on ./webvr-polyfill, hit ENTER to go!: 
./webvr-polyfill does not have GHActions directory
Working on ./webvr-polyfill-dpdb, hit ENTER to go!: 
./webvr-polyfill-dpdb does not have GHActions directory
Working on ./webvrrocks, hit ENTER to go!: 
./webvrrocks does not have GHActions directory
Working on ./webxr, hit ENTER to go!: 
On branch main
Your branch is up to date with 'himorin/main'.

Untracked files:
        index.bs.orig
        index.html
        temp.html

nothing added to commit but untracked files present
Commit failed for ./webxr
Working on ./webxr-ar-module, hit ENTER to go!: 
[main 4f43b58] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/webxr-ar-module
   1f5515e..4f43b58  main -> main
Finished for ./webxr-ar-module
Working on ./webxr-gamepads-module, hit ENTER to go!: 
[main f07e756] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/webxr-gamepads-module
   f99e7c3..f07e756  main -> main
Finished for ./webxr-gamepads-module
Working on ./webxr-hand-input, hit ENTER to go!: 
[main a365247] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/webxr-hand-input
   f2a8858..a365247  main -> main
Finished for ./webxr-hand-input
Working on ./webxr-input-profiles, hit ENTER to go!: 
./webxr-input-profiles does not have GHActions directory
Working on ./webxr-polyfill, hit ENTER to go!: 
./webxr-polyfill does not have GHActions directory
Working on ./webxr-samples, hit ENTER to go!: 
./webxr-samples does not have GHActions directory
Working on ./webxr-test-api, hit ENTER to go!: 
[main 47ee9d2] Updated GHActions file see https://github.com/immersive-web/administrivia/issues/151
 1 file changed, 1 insertion(+), 2 deletions(-)
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 1.10 KiB | 187.00 KiB/s, done.
Total 5 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:immersive-web/webxr-test-api
   0a11b48..47ee9d2  main -> main
Finished for ./webxr-test-api
```

Failure on webxr was due to test mentioned in an initial comment of this issue, fixed by hand operation.

If you get any issue, please reopen this issue.

**Comment by bialpio** [comment]
Hi @himorin, thanks for updating the deploy script! In the future, could you please send a pull request? I have modified the deploy.yml file in the depth-sensing repo to only deploy bikeshed build artifacts (index.html and the img/ folder used by the specification) into gh-pages branch instead of the entire repository, and it seems that your script paved over those changes. We could also come up with some way to manage the repos automatically, but issuing a PR would at least reduce the surprise that I had when I navigated to https://immersive-web.github.io/depth-sensing/. :)

**Comment by himorin** [comment]
@bialpio Thank you for pointing this. Actually opening and merging large number of PRs are huge task since we have number of repositories as above, and we took a way of directly commit to the default branch. 

WG/CG concluded that we don't need a standardized way/manual for setting up repositories, but considering this kind of operation, I suppose we might be better to have some standard (or well considered script(s)) on this area.
Also let me check depth-sensing repository, and push maintenance PR if we can have safe boundary(-es) among maintenance/build files not to cause this kind of issue again.

**Comment by bialpio** [comment]
I think the main problem with using the same GHAction for every repo is that it assumes that the outputs of a Makefile will be produced in-place alongside the rest of the repository. For depth sensing (and I think for hit-test & anchors) I used a Makefile that produced output to a dedicated out/ directory & published just that. I will roll back my changes to the Makefile in depth-sensing repo (others have been modified during migration to GHActions and I have not touched them since, so they should still work) so that it is easier to manage the large amount of repos using the existing scripts, but it would be great if all assumptions that the automated scripts make were listed somewhere, as they may be broken by other repos down the line. For example, real world geometry repo may end up having 2 .bs files since we used it for incubating plane detection and mesh reconstruction - using a Makefile that expects an index.bs will not work there.
